### DISCUSSION TITLE
Wie prüfe ich, ob Besucher im Backend angemeldet ist?

### DISCUSSION THREAD

**Discussion by tyrant88** [FriendsOfREDAXO/.github#2] [Q&A] [answered]
Wie prüfe ich, ob ein Frontend-Aufrufer im Backend angemeldet ist?
Ich möchte im Template einen redirect machen.

genügt `!is_null(rex::getUser())` ?

**Comment by skerbis** [comment]
Ich würde auf 
`rex_backend_login::hasSession() `
prüfen. 

Siehe auch hier: 
https://github.com/FriendsOfREDAXO/maintenance/blob/b35b21da641d79df7cd3e75ac311c4c103aae788/boot.php#L51

**Reply by tyrant88** [reply]
irgendwelche Argumente?

**Reply by skerbis** [reply]
rex::getUser() liefert bei mir im Frontend immer null

**Reply by tyrant88** [reply]
bei mir nicht... sonst wäre ich ja nicht drauf gekommen. Hatte nur noch in Erinnerung, das das nicht Best Practice war.

**Comment by gharlan** [comment, marked as answer]
Wenn man nur wissen möchte, **ob** jemand eingeloggt ist, dann die von @skerbis genannte Methode `rex_backend_login::hasSession()`. Die ist besonders effektiv.

Wenn man auch wissen möchte, **wer** eingeloggt ist (z.b. um zu prüfen, ob Admin), dann mit `rex_backend_login::createUser()` arbeiten.
Die Methode erzeugt das User-Objekt, falls es noch nicht besteht. Im Frontend wird es nicht automatisch erstellt.
Die `hasSession`-Methode muss man dann nicht zusätzlich aufrufen, sie wird intern von `createUser` schon geprüft.

`rex::getUser()` funktioniert, sobald irgendwo `rex_backend_login::createUser()` aufgerufen wurde. Somit kann man sich im Frontend auf `rex::getUser()` nicht verlassen, außer man hat selbst vorher schon mal `createUser` aufgerufen.

**Reply by tyrant88** [reply]
mit "ob jemand" meinst du aber nicht "irgendjemand", sondern genau der User?

**Reply by tyrant88** [reply]
@skerbis müsste ich als Fragesteller nicht entscheiden, was die korrekte Antwort ist?

**Reply by gharlan** [reply]
Achso ja, ist missverständlich. Also mit `hasSession` prüft man, ob der Aufrufer im Backend eingeloggt ist (als irgendein User).
Und mit `createUser` prüft man, (ob und) als welcher Backenduser der Aufrufer eingeloggt ist.

**Reply by skerbis** [reply]
> @skerbis müsste ich als Fragesteller nicht entscheiden, was die korrekte Antwort ist?

Ja. Stimmt. Ich habe den Button falsch verstanden

**Reply by tyrant88** [reply]
Okay, jetzt haben wir genug Wissen angehäuft denke ich. :-)
